### ISSUE TITLE
Tabs, TaNav, TaTitle and Tab - Provide a valid label for form fields - (2036570286)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5605]
### Module: 
 32 Tabs, Tab Nav, Tab Title and Tab 

### Violation: 
 Provide a valid label for form fields 

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570286 

### Severity: 
 10 

### Description: 
 [Issue]
There are tabs after "Test 4. Tabs with no text heading without accessible names. Examples include:

- Rain
- Snowy rain
- Thunderstorms

[User Impact]
Screen reader users will be unable to determine the purpose of these buttons. Speech input users will have difficulty activating them.

[Code Reference] 
 ```
<calcite-tab-title icon-start="rain" aria-selected="false" id="calcite-tab-title-af4c882e-8899-59cb-dea1-cf6bc2da1fc5" role="tab" tabindex="-1" layout="inline" position="top" scale="m" calcite-hydrated="" aria-controls="calcite-tab-title-c6b12df2-2ac7-0c36-5cf3-cc2fcaed62a8"></calcite-tab-title>
 ``` 

### Note: 
 [Suggestion]
Provide a valid label for form fields. This includes providing accessible names for tabs. The accessible name of a tab can be set with an aria-label attribute. Good accessible names are both concise and descriptive.

[Compliant Code Example] 
 ```
<calcite-tab-title aria-label="rain" icon-start="rain" aria-selected="false" id="calcite-tab-title-af4c882e-8899-59cb-dea1-cf6bc2da1fc5" role="tab" tabindex="-1" layout="inline" position="top" scale="m" calcite-hydrated="" aria-controls="calcite-tab-title-c6b12df2-2ac7-0c36-5cf3-cc2fcaed62a8"></calcite-tab-title>
 ```
 

### Media Type: 
 Forms 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Closing this issue as we'll need to document folks should not create `tabs` with only icons present, to ensure proper context to more audiences. Will add an a11y sample to provide context. cc #5606 #5607
